packages/ngtools/webpack/src/paths-plugin.ts

```diff
@@ -112,63 +153,23 @@ export function resolveWithPaths(
     }
   });
 
-  if (pathMapOptions[0].potentials.length === 1) {
-    const onlyPotential = pathMapOptions[0].potentials[0];
-    let replacement;
-    const starIndex = onlyPotential.indexOf('*');
-    if (starIndex === -1) {
-      replacement = onlyPotential;
-    } else if (starIndex === onlyPotential.length - 1) {
-      replacement = onlyPotential.slice(0, -1) + pathMapOptions[0].partial;
-    } else {
-      const [prefix, suffix] = onlyPotential.split('*');
-      replacement = prefix + pathMapOptions[0].partial + suffix;
-    }
-
-    request.request = path.resolve(compilerOptions.baseUrl || '', replacement);
-    callback(null, request);
-
-    return;
-  }
-
-  // TODO: The following is used when there is more than one potential and will not be
-  //       needed once this is turned into a full webpack resolver plugin
-
-  const moduleResolver = ts.resolveModuleName(
```

Comment:
So we give up on TypeScript resolution, which fixed a load of issues and supports tsconfig's path mapping. See https://github.com/angular/angular-cli/pull/7473

Reply:
I'm mostly curious of the two following assumptions in this PR:

1. we implement TypeScript's resolution ourselves, without any bugs, and
1. we rely on TypeScript never updating their resolution algorithm, otherwise this will likely break.

Reply:
Another example is how we lose support for `rootDirs`, which I think we don't test for.

Reply:
`rootDirs` never really worked properly (if ever).  It is also a fairly esoteric option and far more useful outside of bundling scenarios.  However, we could also add support for it.  The logic is very similar to the paths option but for relative imports.

#7473 actually demonstrates some of the issues with using `resolveModuleName`.  A large and ever growing list of special processing is needed to attempt to determine and further locate the correct file to be bundled.  As opposed to the correct file to provide type information for TypeScript transpilation.

Of note also is that the other two typescript loaders (`ts-loader`/`awesome-typescript-loader`) both take the same general approach of re-implementing the algorithm.
In regards to breakage, if TypeScript changed the algorithm or configuration options, it would result in a massive breaking change in regards to TypeScript projects in general.  Failure of the TypeScript team to notify the user base of such a pending change in advance would result in a large percentage of existing TypeScript projects to fail.  Advanced notification would mitigate the danger inherent in such a scenario.

Overall, the more ideal scenario would be if the underlying intrinsics for `resolveModuleName` within TypeScript were made public (someone alterations may also be necessary to allow for full re-usability;  this would need further analysis).  This could potentially allow this plugin to leverage more of the existing logic within TypeScript and limit the need to re-implement.  However, the full Webpack resolver plugin approach would still be the appropriate implementation option as it allows multiple resolution attempts and